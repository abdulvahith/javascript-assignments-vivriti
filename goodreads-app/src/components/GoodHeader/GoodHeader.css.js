export default {
  root: {
    flexGrow: 1,
    width: '100%'
  },
  input: {
    color: "white"
  }
}
