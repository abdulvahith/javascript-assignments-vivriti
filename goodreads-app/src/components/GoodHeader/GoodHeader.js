import React, { Component } from 'react';
import PropTypes from "prop-types";
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { fade, makeStyles } from '@material-ui/core/styles';
import SearchIcon from '@material-ui/icons/Search';
import Button from '@material-ui/core/Button';
import TextField from "@material-ui/core/TextField";
import InputAdornment from "@material-ui/core/InputAdornment";
import CloseIcon from '@material-ui/icons/Close';
import Styles from "./GoodHeader.css.js";

export default class GoodHeader extends Component {
  constructor(props) {
    super(props);
    this.state= {
        userInput: ''
    }
    this.searchBook= this.searchBook.bind(this);
    this.onChange= this.onChange.bind(this);
    this.clearSeach= this.clearSeach.bind(this);
    this.removeSearch= this.removeSearch.bind(this);
  }

  onChange(e){
    this.setState({
      userInput: e.target.value
    })
  }

  searchBook(){
    let {searchBook}= this.props;
    searchBook(this.state.userInput);
  }

  clearSeach(){
    this.setState({
      userInput: ""
    })
  }

  removeSearch(){
    let { getBookList }= this.props;
    this.clearSeach();
    getBookList();
  }

  render() {
    let {isSearch}= this.props;
    return (
      <div style={Styles.root}>
        <AppBar position="static">
          <Toolbar>
            <Typography style={Styles.title} variant="h6" noWrap>
              Good-Read App
            </Typography>
            <div style={{ "marginLeft": "30px", "marginRight": "30px" }}>
              <TextField
                type="search"
                variant="outlined"
                margin="normal"
                value= {this.state.userInput}
                onChange={this.onChange}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <SearchIcon />
                    </InputAdornment>
                  ),
                  endAdornment: (
                    this.state.userInput ?
                    <InputAdornment position="end" style={{"cursor": "pointer"}} onClick={this.clearSeach}>
                      <CloseIcon />
                    </InputAdornment>: null
                  ),
                  style: Styles.input
                }}
              />
            </div>
            <div >
              <Button variant="contained" onClick={this.searchBook} id="search-book"> Search Book </Button>
            </div>
            {
              isSearch?
              <div style={{"marginLeft": "10px", "marginRight": "10px"}}>
                <Button variant="contained" onClick={this.removeSearch} id="clear-search"> Clear search </Button>
              </div>: null
            }

          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

GoodHeader.propTypes = {
  bookList: PropTypes.array,
  searchBook: PropTypes.func,
  getBookList: PropTypes.func,
  isSearch: PropTypes.bool
}
