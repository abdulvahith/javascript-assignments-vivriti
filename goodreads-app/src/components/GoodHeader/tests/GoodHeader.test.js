import React from 'react';
import renderer from 'react-test-renderer';
import GoodHeader from '../GoodHeader';
import { mount, shallow } from 'enzyme';

describe('GoodHeader component should render properly', () => {
    it('renders without crashing', () => {
      shallow(<GoodHeader />);
    });

    it('renders correctly when issearch false snapshot test', () => {
     const rendered = renderer.create(
       <GoodHeader
         isSearch= {false}
       />
     );
     expect(rendered.toJSON()).toMatchSnapshot();
   });

   it('renders correctly when issearch false snapshot test', () => {
    const rendered = renderer.create(
      <GoodHeader
        isSearch= {true}
      />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('search book test', () => {
  const searchBook = jest.fn();

   const wrapper = shallow(
     <GoodHeader
       isSearch= {false}
       searchBook= {searchBook}
     />
   );

   const searchBtn = wrapper.find('#search-book');
   searchBtn.simulate('click');
   expect(searchBook).toHaveBeenCalled();
 });

   it('clear search test', () => {
   const getBookList = jest.fn();

    const wrapper = shallow(
      <GoodHeader
        isSearch= {true}
        getBookList= {getBookList}
      />
    );

    const searchBtn = wrapper.find('#clear-search');
    searchBtn.simulate('click');
    expect(getBookList).toHaveBeenCalled();
  });


});
