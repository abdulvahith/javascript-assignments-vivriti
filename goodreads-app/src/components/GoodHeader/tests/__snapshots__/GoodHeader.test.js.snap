// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoodHeader component should render properly renders correctly when issearch false snapshot test 1`] = `
<div
  style={
    Object {
      "flexGrow": 1,
      "width": "100%",
    }
  }
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
      >
        Good-Read App
      </h6>
      <div
        style={
          Object {
            "marginLeft": "30px",
            "marginRight": "30px",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputTypeSearch"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="search"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-128 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-129"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained"
          disabled={false}
          id="search-book"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
             Search Book 
          </span>
        </button>
      </div>
    </div>
  </header>
</div>
`;

exports[`GoodHeader component should render properly renders correctly when issearch false snapshot test 2`] = `
<div
  style={
    Object {
      "flexGrow": 1,
      "width": "100%",
    }
  }
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
      >
        Good-Read App
      </h6>
      <div
        style={
          Object {
            "marginLeft": "30px",
            "marginRight": "30px",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputTypeSearch"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="search"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-128 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-129"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained"
          disabled={false}
          id="search-book"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
             Search Book 
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        style={
          Object {
            "marginLeft": "10px",
            "marginRight": "10px",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained"
          disabled={false}
          id="clear-search"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
             Clear search 
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
